libs/sql-parser/src/main/java/io/crate/sql/Identifiers.java

```diff
@@ -132,13 +132,25 @@ private static boolean charIsOutsideSafeRange(int i, char c) {
     }
 
     private static boolean quotesRequired(String identifier) {
-        return isKeyWord(identifier) || !IDENTIFIER.matcher(identifier).matches();
+        return isKeyWord(identifier) ||
+               // schema names like s, S, "s" are resolved to s (lower case), while "S" (quoted, mixed case) is resolved to S (upper case).
+               // Therefore, mixed case schema names should be quoted whenever necessary. (i.e. printed as a string and to be re-parsed)
+               !IDENTIFIER.matcher(identifier).matches() || isMixedCase(identifier);
     }
 
     public static boolean isKeyWord(String identifier) {
         return RESERVED_KEYWORDS.contains(identifier.toUpperCase(Locale.ENGLISH));
     }
 
+    private static boolean isMixedCase(String identifier) {
```

Comment:
How about starting with `_`?  doesn't this require quotes?

Reply:
@matriv Please see my comment above, I think we can leave the scope here on uppercase and follow up to support `_` or even all characters.

Reply:
It's totally fine, apologies, we can followup with the rest.